lms/static/scripts/frontend_apps/components/LMSFilePicker.js

```diff
@@ -157,25 +203,37 @@ export default function LMSFilePicker({
         });
       }
     }
-  }, [authToken, listFilesApi, authorizationAttempted]);
+  }, [authToken, folderPath, authorizationAttempted, listFilesApi]);
 
-  // On the initial load, fetch files or prompt to authorize if we know that
-  // authorization will be required.
+  // Update the file list any time the path changes
   useEffect(() => {
     fetchFiles();
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [folderPath]);
 
-  const useSelectedFile = () =>
-    onSelectFile(/** @type {File} */ (selectedFile));
+  /**
+   *
+   * @param {File|null} [file]
+   * @returns
+   */
+  const confirmSelectedItem = (file = selectedFile) => {
```

Comment:
```suggestion
  /** @param {File|null} file */
  const confirmSelectedItem = (file = selectedFile) => {
```

In cases where the JSDoc is just adding type info, I prefer to keep it succinct and avoid extraneous lines. This makes the code easier to skim as a whole.